lib/runtime/modules/array.js

```diff
@@ -28,7 +38,116 @@ var array = {
         }
 
         return index;
-    }
+    },
+    join: function(array, joiner) {
+        if (!values.isArray(array)) {
+            throw errors.typeErrorFunction('Array.join', 'array', array);
+        }
+        if (!values.isString(joiner)) {
+            throw errors.typeErrorFunction('Array.join', 'string', joiner);
+        }
+
+        return array.join(joiner);
+    },
+    lastIndexOf: function(array, searchValue) {
+        if (!values.isArray(array)) {
+            throw errors.typeErrorFunction('Array.lastIndexOf', 'array', array);
+        }
+
+        var index = -1;
+
+        for (var i = array.length - 1; i > 0; i--) {
+            if (values.equal(array[i],searchValue)) {
+                index = i;
+                break;
+            }
+        }
+
+        return index;
+    },
+    pop: function(array) {
+        if (!values.isArray(array)) {
+            throw errors.typeErrorFunction('Array.pop', 'array', array);
+        }
+
+        return array.pop();
```

Comment:
On empty array, this would return `undefined`. We need to check for that and return `null` instead.